File: src/main/java/com/example/demo/javadomain/Circle.java
package com.example.demo.javadomain;

public class Circle implements Shape {
    public final double radius;
    public final String color;

    public Circle(double radius, String color){
        this.radius = radius;
        this.color = color;
    }

    @Override
    public String getColor() { return this.color; }

    @Override
    public double getArea() { return Math.PI * Math.pow(radius, 2); }

    @Override
    public double getPerimeter() { return 2 * Math.PI * radius; }
}

File: src/main/kotlin/com/example/demo/kotlindomain/Circle.kt
package com.example.demo.kotlindomain

import kotlin.math.pow

class Circle(private val radius: Double, override val color: String? = null) : Shape {
    override val area: Double
        get() = Math.PI * radius.pow(2.0)

    override val perimeter: Double
        get() = 2 * Math.PI * radius
}